ACPs/181-proposervm-epochs/README.md

```diff
@@ -0,0 +1,141 @@
+
+## Specification
+
+### Assumptions
+
+In the following specification, we assume that a block $b_m$ has timestamp $t_m$ and P-Chain height $p_m$.
+
+### Epoch Definition
+
+An epoch is defined as a contiguous range of blocks that share the same three values:
+- An Epoch Number
+- An Epoch P-Chain Height
+- An Epoch Start Time
+
+Let $E_N$ denote an epoch with epoch number $N$. $E_N$'s start time is denoted as $T_{start}^N$, and its P-Chain height as $P_N$. 
+
+$E_0$ is defined as the epoch whose start time $T_{start}^0$ is the block timestamp of the block that activates this ACP (i.e. the first block at or following this ACP's activation timestamp).
+
+### Epoch Sealing
+
+An epoch $E_N$ is *sealed* by the first block with a timestamp greater than or equal to $T_{start}^N + D$, where $D$ is a constant defined in the network upgrade that activates this ACP. Let $B_{S_N}$ denote the block that sealed $E_N$.
+
+The sealing block is defined to be a member of the epoch it seals. This guarantees that every epoch will contain at least one block. 
+
+### Advancing an Epoch
+
+We advance from the current epoch $E_N$ to the next epoch $E_{N+1}$ when the next block after $B_{S_N}$ is produced. This block will be a member of $E_{N+1}$, and will have the values:
+- $P_{N+1}$ equal to the P-Chain height of $B_{S_N}$
+- $T_{start}^{N+1}$ equal to $B_{S_N}$'s timestamp
+- The epoch number, $N+1$ increments the previous epoch's epoch number by exactly $1$
+
+## Properties and Use Cases
+
+### Epoch Duration Bounds
+
+Since an epoch's start time is set to the [timestamp of the sealing block of the previous epoch](#advancing-an-epoch), all epochs are guaranteed to have a duration of at least $D$, as measured from the epoch's starting time to the timestamp of the epoch's sealing block. However, since a sealing block is [defined](#epoch-sealing) to be a member of the epoch it seals, there is no upper bound on an epoch's duration, since that sealing block may be produced at any point in the future beyond $T_{start}^N + D$.
+
+### Fixing the P-Chain Height
+
+When building a block, Avalanche blockchains use the P-Chain height [embedded in the block](#assumptions) to determine the validator set. If instead the epoch P-Chain height is used, then we can ensure that when a block is built, the validator set to be used for the next block is known. To see this, suppose block $b_m$ seals epoch $E_N$. Then the next block, $b_{m+1}$ will begin a new epoch, $E_{N+1}$ with $P_{N+1}$ equal to $b_m$'s P-Chain height, $p_m$. If instead $b_m$ does not seal $E_N$, then $b_{m+1}$ will continue to use $P_{N}$. Both candidates for $b_{m+1}$'s P-Chain height ($p_m$ and $P_N$) are known at $b_m$ build time.
+
+## Backwards Compatibility
+
+This change requires a network upgrade and is therefore not backwards compatible.
```

Comment:
I'd be great to add a second paragraph here calling out any known incompatibilities this upgrade will cause. The only one I can think of right now is that relayers of ICM message will need to account for the destination chain's epoching scheme in order to be able to construct an aggregate signature that will be verified properly